### PR TITLE
No empty strings in to_dict

### LINKED ISSUE #460: User.to_dict() returns absent fields with empty strings '' values
### Steps to reproduce
1. Register any bot in Telegram. Here I will use one with smudged sensitive information.
Get its own information using `getMe` telegram api call.
For example (output pretty-printed):
```
$ http https://api.telegram.org/bot17XXXXX3:AXXcXnXXXXXXXXQgcXXXXc/getMe
HTTP/1.1 200 OK
Access-Control-Allow-Methods: GET, POST, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: Content-Length,Content-Type,Date,Server,Connection
Connection: keep-alive
Content-Length: 89
Content-Type: application/json
Date: Sun, 06 Nov 2016 22:00:54 GMT
Server: nginx/1.10.1
Strict-Transport-Security: max-age=31536000; includeSubdomains

{
    "ok": true,
    "result": {
        "first_name": "test",
        "id": 17XXXXX3,
        "username": "test_bot"
    }
}
```

2. Try to get the same information using Python Telegram Bot.
Version 3.4.0 (prior to 4.0.0):
```
$ bpython
bpython version 0.16 on top of Python 3.5.1 /Users/el/.pyenv/versions/3.5.1/bin/python3.5
>>> import telegram
>>> bot=telegram.Bot('bot17XXXXX3:AXXcXnXXXXXXXXQgcXXXXc')
>>> user=bot.getMe()
>>> user.to_dict()
{'first_name': 'test', 'id': 17XXXXX3, 'username': 'test_bot'}
```

Version after 4+ (for example 5.2.0):
```
$ bpython
bpython version 0.16 on top of Python 3.5.1 /Users/el/.pyenv/versions/3.5.1/bin/python3.5
>>> import telegram
>>> bot=telegram.Bot('bot17XXXXX3:AXXcXnXXXXXXXXQgcXXXXc')
>>> user=bot.getMe()
>>> user.to_dict()
{'first_name': 'test', 'last_name': '', 'type': '', 'id': 17XXXXX3, 'username': 'test_bot'}```
```

3. 

### Expected behaviour
Should return only fields got with Telegram API (id, first_name, username), do not add empty optional fields (last_name, type).

### Actual behaviour
Returned dict is populated with absent optional fields (last_name, type) with empty string values.

### Configuration
**Operating System:**
MaxOS 10.12.1 
Also tested on Linux boxes in different CI platforms (like Travis-CI, Codeship)

**Version of Python, python-telegram-bot & dependencies:**
Reproduced on 3.4.5, 3.5.1, 3.5.2, 3.6b4+

```
$ python -m telegram
/Users/el/.pyenv/versions/3.5.1/bin/python: No module named telegram.__main__; 'telegram' is a package and cannot be directly executed
```
### Logs
Insert logs here (if necessary)

Issue was found after upgrading python-telegram-bot from 3.2 to 5.2.0 used as an intermediate dependency in my project.

### Reason
One of the reasons to emerge the issue can be a [changes](https://github.com/python-telegram-bot/python-telegram-bot/compare/v3.2...v5.2.0?diff=split&name=v5.2.0#diff-4a13231a9fa379318457405e5df4712bL63) between versions. While dumping user data to dict, old version ignored empty field checking them (``if field: ...``) thus 0, false and some other values could be lost. New version checks them (``if field is not None: ...``) in this case we ignore only real empty (``None``) fields, but method ``User.__init__`` [uses empty string as default values](https://github.com/python-telegram-bot/python-telegram-bot/compare/v3.2...v5.2.0?diff=split&name=v5.2.0#diff-c9df4cc0aa3986f1f87042b987ec5bd4R47) for absent parameters. So User object gets ``''`` as values.

## Possible impact
Can break some other products dependant on returned values of ``getMe`` method.

## Proposition
Change signature of ``User`` constructor to use ``None`` as default value for absent optional parameters.

If you wish I can make a PR with tests.

### PR DESCRIPTION
**Pull request by bomjacob**
Should remove all cases of having empty strings returned from to_dict (if I didn't miss any).

Closes #460

### CODE REVIEW

File: telegram/contact.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ class Contact(TelegramObject):
 
     """
 
-    def __init__(self, phone_number, first_name, last_name='', user_id=0, **kwargs):
+    def __init__(self, phone_number, first_name, last_name=None, user_id=0, **kwargs):
```
Comment (tsnoam) [inline telegram/contact.py]: I think `user_id` should be default to `None`, same reasons like with `Chat` above.

File: telegram/document.py [hunk 2]
```diff
@@ -41,13 +41,14 @@ class Document(TelegramObject):
...
-    def __init__(self, file_id, thumb=None, file_name='', mime_type='', file_size=0, **kwargs):
+    def __init__(self, file_id, thumb=None, file_name=None, mime_type=None, file_size=0, **kwargs):
         # Required
         self.file_id = str(file_id)
         # Optionals
         self.thumb = thumb
         self.file_name = file_name
-        self.mime_type = str(mime_type)
+        if mime_type:
+            self.mime_type = str(mime_type)
```
Comment (tsnoam) [inline telegram/document.py]: Please make sure `self.mime_type` is always available. I think that this should be sufficient (no need to use `str()` here):
`self.mime_type = mime_type` 
If you still think `str()` is required, you can do:
`self.mime_type = str(mime_type) if mime_type is not None else None`

File: telegram/audio.py [hunk 3]
```diff
@@ -46,9 +46,9 @@ class Audio(TelegramObject):
     def __init__(self,
                  file_id,
                  duration,
-                 performer='',
-                 title='',
-                 mime_type='',
+                 performer=None,
+                 title=None,
+                 mime_type=None,
                  file_size=0,
```
Comment (bomjacob) [inline telegram/audio.py]: We have that in a few places... I don't really know...
It's the same on with the booleans on Message.

File: telegram/audio.py [hunk 4]
```diff
@@ -57,7 +57,8 @@ def __init__(self,
         # Optionals
         self.performer = performer
         self.title = title
-        self.mime_type = str(mime_type)
+        if mime_type:
```
Comment (bomjacob) [inline telegram/audio.py]: I'mma fix it anyways :)

Comment (bomjacob) [inline telegram/audio.py]: Actually nvm, I saw your comment further down ^^

File: telegram/chat.py [hunk 5]
```diff
@@ -53,11 +53,11 @@ class Chat(TelegramObject):
...
-                 title='',
-                 username='',
-                 first_name='',
-                 last_name='',
-                 all_members_are_admins=False,
+                 title=None,
+                 username=None,
+                 first_name=None,
+                 last_name=None,
+                 all_members_are_admins=None,
```
Comment (tsnoam) [inline telegram/chat.py]: The API of Chat calls this var as `all_members_are_administrators`, here we use `admins`. Is there anything that translates `admins`. I think we actually have a bug here, or telegram might have silently changed the API.

File: telegram/file.py [hunk 6]
```diff
@@ -40,13 +40,14 @@ class File(TelegramObject):
...
+    def __init__(self, file_id, bot, file_size=None, file_path=None, **kwargs):
         # Required
         self.file_id = str(file_id)
 
         # Optionals
-        self.file_size = int(file_size)
-        self.file_path = str(file_path)
+        self.file_size = file_size
+        if file_path:
+            self.file_path = str(file_path)
```
Comment (tsnoam) [inline telegram/file.py]: Just do
`self.file_path = file_path`
The `if` statement should not be there.

File: telegram/user.py [hunk 7]
```diff
@@ -44,7 +44,14 @@ class User(TelegramObject):
         bot (Optional[Bot]): The Bot to use for instance methods
     """
 
-    def __init__(self, id, first_name, type='', last_name='', username='', bot=None, **kwargs):
+    def __init__(self,
+                 id,
+                 first_name,
+                 type=None,
```
Comment (tsnoam) [inline telegram/user.py]: I don't see the `type` attribute in documentation. I think we should remove it too.
I can only guess it's something Telegram had obsoleted.
You should leave it at the prototype until ver. 7 I guess, and put a deprecation warning if it specified.

File: telegram/webhookinfo.py [hunk 8]
```diff
@@ -40,13 +40,20 @@ class WebhookInfo(TelegramObject):
 
     """
 
-    def __init__(self, url, has_custom_certificate, pending_update_count, **kwargs):
+    def __init__(self,
+                 url,
+                 has_custom_certificate,
+                 pending_update_count,
+                 last_error_date=None,
+                 last_error_message=None,
```
Comment (tsnoam) [inline telegram/webhookinfo.py]: We're missing optionals: `max_connections` (`int`) & `allowed_updates` (`List[str]`)

File: telegram/chat.py (hunk 5 above)
Comment (bomjacob) [inline telegram/chat.py]: I'm about 90% sure they changed their API... so should we change it, or like keep both, or?

File: telegram/file.py (hunk 6 above)
Comment (bomjacob) [inline telegram/file.py]: Right, whoops. I removed the other ones. Seems like I missed these.

File: telegram/user.py (hunk 7 above)
Comment (bomjacob) [inline telegram/user.py]: Hmm.. I don't think it's ever been there tbh.. I might be us who made an error and confused User for Chat or something.

File: telegram/webhookinfo.py (hunk 8 above)
Comment (bomjacob) [inline telegram/webhookinfo.py]: I noticed but I thought @jh0ker was working on that?
Now that I look at it, it doesn't seem to be in our PR list... (I think I confused it for #483)

File: telegram/chat.py (hunk 5 above)
Comment (tsnoam) [inline telegram/chat.py]: We should follow their official API. 
We *can* do bkwd-compat for a while using **kwargs.
@jh0ker What do you think?

File: telegram/user.py (hunk 7 above)
Comment (tsnoam) [inline telegram/user.py]: I think Telegram removed it.
@jh0ker what do you think?

File: telegram/webhookinfo.py (hunk 8 above)
Comment (tsnoam) [inline telegram/webhookinfo.py]: I thought that too, so I checked the official December 4th release notes and it wasn't there, so lets fix it here while we're at it.

File: telegram/chat.py (hunk 5 above)
Comment (jh0ker) [inline telegram/chat.py]: I'd say we keep bkwd-compat

File: telegram/user.py (hunk 7 above)
Comment (jh0ker) [inline telegram/user.py]: I think I remember it from the official docs as well, but I could be wrong...

File: telegram/webhookinfo.py (hunk 8 above)
Comment (bomjacob) [inline telegram/webhookinfo.py]: Wait no... it totally is part of #483 isn't it, @jh0ker?

**Pull request merged**